src/video/SDL_pixels.c

```diff
@@ -1235,106 +1235,135 @@ void SDL_DetectPalette(SDL_Palette *pal, SDL_bool *is_opaque, SDL_bool *has_alph
 }
 
 /* Find the opaque pixel value corresponding to an RGB triple */
-Uint32 SDL_MapRGB(const SDL_PixelFormat *format, Uint8 r, Uint8 g, Uint8 b)
+Uint32 SDL_MapRGB(SDL_PixelFormat format, SDL_Palette *palette, Uint8 r, Uint8 g, Uint8 b)
 {
-    if (!format) {
+    if (palette) {
+        return SDL_FindColor(palette, r, g, b, SDL_ALPHA_OPAQUE);
+    }
+
+    SDL_PackedPixelDetails *fmt = SDL_GetPackedPixelDetails(format);
```

Comment:
One concern would be the cost of looking up the pixel format details constantly, though I suppose these functions have never really been performance critical